runtime/vm/jvminit.c

```diff
@@ -7706,6 +7706,46 @@ protectedInitializeJavaVM(J9PortLibrary* portLibrary, void * userData)
 
 	env->gpProtected = FALSE;
 
+	{
+		J9VMInitArgs* args = vm->vmArgsArray;
```

Comment:
Calling into java in this stage is inherently risky.

I suggest the following:
1) Parse the `VMOPT_XXSTARTFLIGHTRECORDING` option in `processVMArgsFromFirstToLast` saving the the option contents to jfrState->jfrCMDLineOption

2) Create a native in `com.ibm.oti.vm.VM.getjfrCMDLineOption()` as a string

3) Add a function called initJFR to java.lang.System which essentially does what you have below but can be done in java code and reflection. If you run into boots strapping issues then you can make a native and use JNI.

4) In java.ang.System.afterClinitInitialization() you can call that function.

Reply:
For 4, If I call initJFR from `afterClinitInitialization` I'm getting CNFE (`java.lang.ClassNotFoundException: jdk.jfr.internal.dcmd.DCmdStart`) as the jdk modules doesn't seem to be initialised at that point. 

Hence I'm calling it in java.lang.System.completeInitialization()

PTAL, Thanks @tajila !